### PR TITLE
New feature: allow spoon to do static imports for methods and fields

### PR DESCRIPTION
**Pull request by surli**
Improve `ImportScannerImpl` to allow Spoon to do static imports. 
Change the interface of `ImportScanner` to return list of CtReference when doing computeImports. 

Lots of tests had to be changed as with autoImports it automatically imports all static elements.

### CODE REVIEW

File: src/main/java/spoon/reflect/visitor/ImportScannerImpl.java [hunk 1]
```diff
@@ -132,134 +146,305 @@ public void scan(CtElement element) {
...
+		listallImports.addAll(this.classImports.values());
+		listallImports.addAll(this.fieldImports.values());
+		listallImports.addAll(this.methodImports.values());
+		return listallImports;
 	}
 
 	@Override
 	public void computeImports(CtElement element) {
-		imports.clear();
+		classImports.clear();
```
Comment (tdurieux) [inline src/main/java/spoon/reflect/visitor/ImportScannerImpl.java]: You have to clear the 3 lists

Comment (surli) [inline src/main/java/spoon/reflect/visitor/ImportScannerImpl.java]: Agreed. I push the change.

**Comment by monperrus** [comment]
cool!

in Spoon, we aim at maximum backward compatibility and not to change the signature public methods, esp. in interfaces. 

to this extent, I propose to keep computeImports as is and to introduce a new method (eg computeAllImports).

this will greatly reduce the impact in the tests (and in client code also).

(fortunately isImported(CtReference ref) is backward compatible)

**Comment by surli** [comment]
> to this extent, I propose to keep computeImports as is and to introduce a new method (eg computeAllImports).
> this will greatly reduce the impact in the tests (and in client code also).

Ok but then DefaultJavaPrettyPrinter will use the computeAllImports, right?

**Comment by surli** [comment]
Ok to merge this now @monperrus ?

**Pull request merged**